src/components/owned/OwnerInfo.tsx

```diff
@@ -29,82 +28,67 @@ const walletBox = classnames(
 )
 const walletAddress = classnames(display('flex'), flexDirection('flex-col'))
 
-function BadgeTitle({ badge }: { badge: BaseBadgeContract }) {
-  if (badge instanceof EmailBadgeContract) {
-    return (
-      <>
-        <HeaderText extraLeading>
-          This wallet belongs to someone with{' '}
-          <ExternalLink
-            url={getEtherscanAddressUrl(badge.address, Network.Goerli)}
-          >
-            <AccentText bold color="text-secondary">
-              <ContractName
-                hyphens
-                address={badge.address}
-                network={Network.Goerli}
-              />
-            </AccentText>
-          </ExternalLink>
-        </HeaderText>
-      </>
+function replace(
+  template: string,
+  keyword: string,
+  element: JSXInternal.Element
+) {
+  return template
+    .split(keyword)
+    .reduce(
+      (chain, part, index) =>
+        index === 0 ? chain.concat(part) : chain.concat(element).concat(part),
+      [] as (JSXInternal.Element | string)[]
     )
-  }
+}
 
-  return (
-    <>
-      <HeaderText extraLeading>
-        <span className={display('md:block')}>This wallet owns </span>a{' '}
-        <ExternalLink
-          url={getEtherscanAddressUrl(badge.address, Network.Goerli)}
-        >
-          <AccentText bold color="text-secondary">
-            <ContractName
-              hyphens
-              address={badge.address}
-              network={Network.Goerli}
-            />
-          </AccentText>
-        </ExternalLink>
-      </HeaderText>
-    </>
+function BadgeTitle({
+  badge: { type, derivative },
+}: {
+  badge: BaseBadgeContract
+}) {
+  const { ownerTitle } = data[type]
+
+  if (!ownerTitle) return null
```

Comment:
Will destructuring work if there's nothing to destructure?

Reply:
Types are supposed to tell if it's not possible, right?